packages/flutter_tools/lib/src/isolated/devfs_web.dart

```diff
@@ -179,6 +180,29 @@ class WebAssetServer implements AssetReader {
     _hotRestartGeneration++;
   }
 
+  /// Given a list of [modules] that need to be reloaded, writes a file that
+  /// contains a list of objects each with two fields:
+  ///
+  /// `src`: A string that corresponds to the file path containing a DDC library
+  /// bundle.
+  /// `libraries`: An array of strings containing the libraries that were
+  /// compiled in `src`.
+  ///
+  /// The path of the output file should stay consistent across the lifetime of
+  /// the app.
+  void performReload(List<String> modules) {
+    final List<Map<String, Object>> moduleToLibrary = <Map<String, Object>>[];
+    for (final String module in modules) {
+      final ModuleMetadata metadata = ModuleMetadata.fromJson(
+        json.decode(utf8.decode(_webMemoryFS.metadataFiles['$module.metadata']!.toList()))
+            as Map<String, dynamic>,
+      );
+      final List<String> libraries = metadata.libraries.keys.toList();
+      moduleToLibrary.add(<String, Object>{'src': module, 'libraries': libraries});
+    }
+    writeFile('main.dart.js.reloadScripts', json.encode(moduleToLibrary));
```

Comment:
Is the "main.dart" portion of this filename based on the hard coded name of a wrapping script? If it is always the same, would something like "reload_scripts.json" make more sense? I also try to keep camel case out of filenames in the dart-y world.

Reply:
So initially I wanted to emulate what we did internally for hot restart by adding a query suffix, but then I was worried that the cache busting wouldn't play nicely with that so I ended up with this.

I agree this would be much nicer just writing the restart and reload scripts like you propose and reading that path later. Done.